app/jobs/in_person/enrollments_ready_for_status_check/batch_processor.rb

```diff
@@ -0,0 +1,75 @@
+module InPerson::EnrollmentsReadyForStatusCheck
+  class BatchProcessor
+    # @param [InPerson::EnrollmentsReadyForStatusCheck::ErrorReporter] error_reporter
+    # @param [InPerson::EnrollmentsReadyForStatusCheck::SqsBatchWrapper] sqs_batch_wrapper
+    # @param [InPerson::EnrollmentsReadyForStatusCheck::EnrollmentPipeline] enrollment_pipeline
+    def initialize(error_reporter:, sqs_batch_wrapper:, enrollment_pipeline:)
+      @error_reporter = error_reporter
+      @sqs_batch_wrapper = sqs_batch_wrapper
+      @enrollment_pipeline = enrollment_pipeline
+    end
+
+    # Process a batch of incoming messages corresponding to in-person
+    # enrollments that are ready to have their status checked.
+    #
+    # Note: Stats are accepted as param to increment and facilitate logging even
+    # if this method raises an error.
+    #
+    # @param [Array<Aws::SQS::Types::Message>] messages In-person enrollment SQS messages
+    # @param [Hash] analytics_stats Counters for aggregating info about how items were processed
+    # @option analytics_stats [Integer] :fetched_items Items received from SQS
+    # @option analytics_stats [Integer] :valid_items Items matching the expected format/data
+    # @option analytics_stats [Integer] :invalid_items Items not matching the expected format/data
+    # @option analytics_stats [Integer] :processed_items Items processed without errors
+    # @option analytics_stats [Integer] :deleted_items Items successfully deleted from queue
```

Comment:
Are these items that encountered errors when we attempted to process them?

Reply:
We're deleting items in the following cases:
- Processing succeeds
- The item is malformed (e.g. invalid JSON or has wrong fields)
- The item does not correspond to a pending enrollment record

If we don't delete the last two, then workers will continue attempting (and failing) to process them.